### PR TITLE
feat: add new model for `HttpHeaders`

### PR DESCRIPTION
**Pull request by vdusek**
### Description

- Unify the HTTP headers type across the project.
- ~I used just the type alias plus custom validator in the `Request` model to reach the same result (lowercase & sorted).~
  - ~Of course, now it is only in the `Request`, but I believe it is not a problem.~
- Take it out from #542.

### Issues

- N/A

### Testing

- N/A

### Checklist

- [x] CI passed

### CODE REVIEW

File: src/crawlee/_request.py [hunk 1]
```diff
@@ -119,7 +120,12 @@ class BaseRequestData(BaseModel):
     method: HttpMethod = 'GET'
     """HTTP request method."""
 
-    headers: Annotated[dict[str, str], Field(default_factory=dict)] = {}
+    headers: Annotated[
+        HttpHeaders,
+        # Normalize headers to lowercase keys and sort them.
+        PlainValidator(lambda value: normalize_headers(value)),
+        Field(default_factory={}),
```
Comment (janbuchar) [inline src/crawlee/_request.py:127]: Shouldn't `default_factory` be a callable? And do we really need it?

**Review by janbuchar — COMMENTED** [review]
What about http client? That uses `HttpHeaders` as well and I'd expect normalization to work there.

File: src/crawlee/_request.py (hunk 1 above)
Comment (vdusek) [reply src/crawlee/_request.py:127]: oh, good catch

File: src/crawlee/_request.py [hunk 2]
```diff
@@ -119,7 +166,7 @@ class BaseRequestData(BaseModel):
     method: HttpMethod = 'GET'
     """HTTP request method."""
 
-    headers: Annotated[dict[str, str], Field(default_factory=dict)] = {}
+    headers: Annotated[HttpHeaders, Field(default_factory=HttpHeaders())] = HttpHeaders()
```
Comment (janbuchar) [inline src/crawlee/_request.py:169]: Suspicious default factory, do we really need both that and a regular default?

Also, doesn't this cause `Request` to serialize to `{..., "headers": {"headers": {...}}, ...}` now? Wouldn't a `RootModel` be more appropriate? :slightly_smiling_face:

File: src/crawlee/_request.py [hunk 3]
```diff
@@ -25,6 +25,53 @@
...
 
 
+def _normalize_headers(headers: Mapping[str, str]) -> dict[str, str]:
+    """Converts all header keys to lowercase and returns them sorted by key."""
+    normalized_headers = {k.lower(): v for k, v in headers.items()}
+    sorted_headers = sorted(normalized_headers.items())
+    return dict(sorted_headers)
+
+
+class HttpHeaders(BaseModel, Mapping[str, str]):
```
Comment (janbuchar) [inline src/crawlee/_request.py:35]: Maybe we could move this to utils, types or something?

Comment (vdusek) [reply src/crawlee/_request.py]: I will leave it in the `_types` then.

File: src/crawlee/_request.py (hunk 2 above)
Comment (vdusek) [reply src/crawlee/_request.py:122]: > Suspicious default factory, do we really need both that and a regular default?

yeah, mypy

> Also, doesn't this cause Request to serialize to {..., "headers": {"headers": {...}}, ...} now? Wouldn't a RootModel be more appropriate? 🙂

Thanks!

**Review by janbuchar — APPROVED** [review]
LGTM

**Pull request merged**